### PR TITLE
finalize w/ systemfonts & ragg

### PR DESCRIPTION
**Pull request by matthewstern**
I'm creating a PR as draft for @sarahcmap's work on updating `finalize_plot()` to work with ragg. Work isn't done yet, but with multiple potential contributors at this point, this will be useful to track what's still to do. The goal is to get plot finalization working with ragg, svglite, and systemfonts. See #127 for context.

Sarah wrote on 5/4 in teams:
> png, tiff, jpeg, and pdf exports look ok.  svg title doesn't appear to be bold, and I still haven't work out the mode='window' situation. 

I able to remove `mode = "window"` and related code, but didn't get to look at svg outputs.

### CODE REVIEW

File: vignettes/finalize.Rmd [hunk 1]
```diff
@@ -56,7 +56,7 @@ In this vignette we will use the final version of the line chart developed in `v
 
 After creating a plot and applying `theme_cmap()`, use `finalize_plot()` to complete the implementation of CMAP design standards. You will probably want to set at least the `title` and `caption`, although the function will extract them from the ggplot if they were specified. 
 
-As you are preparing the plot, you will likely want to view it within R. Do this by leaving leaving the default `mode = "plot"` to send the finished plot to the "Plots" tab within RStudio. Windows users may prefer `mode = "window"`, which draws the plot in a popup window. In both modes, the plot will show up "actual size" (depending on your screen's resolution) surrounded by a grey canvas.
+As you are preparing the plot, you will likely want to view it within R. Do this by leaving leaving the default `mode = "plot"` to send the finished plot to the "Plots" tab within RStudio. The plot will show up "actual size" (depending on your screen's resolution) surrounded by a gray canvas.
```
Comment (nmpeterson) [inline vignettes/finalize.Rmd:59]: Should we add a note about clicking "zoom" to open a plot in a window, especially when the plots pane is too small to show the entire plot?

**Comment by matthewstern** [comment]
@sarahcmap, one thought is whether the svglite package is up to date on your machine? Presumably the package was updated to use systemfonts at some point relatively recently. 

In fact, we may want to require a certain version of that package, presuming older versions don't handle fonts correctly and new ones do?

File: vignettes/finalize.Rmd (hunk 1 above)
Comment (dlcomeaux) [reply vignettes/finalize.Rmd]: I've added a couple lines accordingly.

**Comment by dlcomeaux** [comment]
OK. @matthewstern and I think that at this stage, it might make sense to move ahead and use the original `svg` device. There appear to be some underlying problems in how Whitney, as registered, is being interpreted by the `svglite` package, as discussed [here](https://github.com/CMAP-REPOS/cmapplot/issues/127#issuecomment-853339664). In the interests of getting the rest of 1.2 implemented, I suggest that we merge this in its current form into 1.2 so that we can bring that into the master.

I will create a consolidated issue for a future enhancement to shift from `svg` to `svglite`, and/or to at least add `svglite` as an option (per Noel's suggestion [here](https://github.com/CMAP-REPOS/cmapplot/issues/91#issuecomment-738859149)).

@sarahcmap since you've been working on this, are you OK with that approach? @nmpeterson , any concerns from you?

File: R/finalize_plot.R [hunk 2]
```diff
@@ -4,9 +4,8 @@
 #'your title and caption to the left, add a horizontal line on top, and make
 #'other adjustments. It can show you the final plot and/or export it as a raster
 #'or vector file. This function will not apply CMAP design standards to the plot
-#'itself: use with \code{theme_cmap()} for that. Exports from this function use
-#'Cairo graphics drivers, while drawing within R is done with default (Windows)
-#'drivers.
+#'itself: use with \code{theme_cmap()} for that. This function uses ragg drivers
+#'in R and for raster exports, svglite for svg, and cairo_pdf for PDFs.
```
Comment (matthewstern) [inline R/finalize_plot.R:8]: needs updating

**Review by matthewstern — COMMENTED** [review]
amused that I can't "approve" a PR that I opened, but this looks good to me. I'll do more testing on the larger 1.2 branch once this is merged in, but all the code changes seem fine to me.

File: R/finalize_plot.R (hunk 2 above)
Comment (dlcomeaux) [reply R/finalize_plot.R:8]: OK, done.

**Comment by sarahcmap** [comment]
Thanks @dlcomeaux and @matthewstern for troubleshooting!  I think your plan sounds fine Daniel.

**Pull request merged**